### DISCUSSION TITLE
How to add application logs to spans

### DISCUSSION THREAD

**Discussion by Suedo** [open-telemetry/opentelemetry-java-instrumentation#2969] [Q&A] [answered]
Below is my auto-instrumentation config:

```js
-Dotel.traces.exporter=jaeger
-Dotel.metrics.exporter=none
-Dotel.exporter.jaeger.endpoint=http://localhost:14250
-Dotel.resource.attributes=service.name=myService
-javaagent:C:/path/to/opentelemetry-javaagent-1.0.1-all.jar
```

I am using Jaegar as the backend, and in the trace that comes up, I would want to add application logs (we use logback) to the spans. Is that possible?

**Comment by trask** [comment]
hey @Suedo! OpenTelemetry focus was initially on tracing, now on metrics, and after that logging, so we are definitely planning to address this use case, but don't have anything in instrumentation yet (OpenTelemetry logging is still going through specs, then SDK, then instrumentation).

Currently, the only thing that the instrumentation supports around logging is https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/logger-mdc-instrumentation.md.

**Reply by Suedo** [reply]
@trask thanks for the details. This is a silly question, but asking just in case..

Is there a way for OpenTelemetry auto-instrumentation to work with Jaegar tracer ? I know [Jaegar spans supports logs](https://github.com/jaegertracing/jaeger-client-java/discussions/786#discussioncomment-748549), and would be really nice if I can leverage that with auto-instrumentation, atleast till OpenTemetry comes with with logging capabilities.

**Reply by trask** [reply]
@Suedo there's https://github.com/open-telemetry/opentelemetry-java/tree/main/opentracing-shim

you could also probably implement your own logback appender that adds logs as events to the current span

**Reply by itsprakash84** [reply]
@Suedo ,

I do have same requirement, any luck on customer logback appender? if yes so please point me there.

@trask ,
Do you have roadmap for logback/log4j auto instrumentation support?

**Reply by trask** [reply]
hey @itsprakash84! check out and follow #3055

**Reply by Suedo** [reply]
Hey @trask , while playing with OTEL + Jaegar, I saw that `currentSpan.recordException(throwable)` results in a `Log` section in the span. Is there a way to make this Log section appear for non-exception cases also?

![image](https://user-images.githubusercontent.com/4963779/128864641-09a2063d-322c-4554-a725-1e15ccfb3423.png)

ps; office machine, so had to redact most stuff, sorry for the ugliness

**Reply by iNikem** [reply]
Exceptions are recorded as span events. I believe jaeger exporter converts them to Jaeger logs. You can add arbitrary events to spans via Otel API.

**Reply by Suedo** [reply]
Awesome! This works really well! I followed this [span event guide](https://opentelemetry.io/docs/java/manual_instrumentation/#create-spans-with-events) and I can see the details in a Log section! @iNikem can you mention this as a separate answer, then I can mark that one as the fix

@itsprakash84 take a look, u might find this useful

**Comment by iNikem** [comment, marked as answer]
Exceptions are recorded as span events. I believe jaeger exporter converts them to Jaeger logs. You can add arbitrary events to spans via Otel API.

**Comment by itsprakash84** [comment]
Thank you @Somjit Nag ***@***.***> , I have implemented this
sometime back.

BTW, has anyone tried exporting custom metrics using otel?


On Wed, Aug 11, 2021 at 2:58 PM Nikita Salnikov-Tarnovski <
***@***.***> wrote:

> Exceptions are recorded as span events. I believe jaeger exporter converts
> them to Jaeger logs. You can add arbitrary events to spans via Otel API.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/open-telemetry/opentelemetry-java-instrumentation/discussions/2969#discussioncomment-1157801>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABQ4WXPXU3IN2US2UXQYHTTT4I7FHANCNFSM44YA6VKA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>


-- 
*Regards,*
*Sathya Prakash*
